src/test/java/net/sf/jsqlparser/statement/merge/MergeTest.java

```diff
@@ -126,17 +125,6 @@ public void testMergeUpdateInsertOrderIssue401_2() throws JSQLParserException {
                 "MERGE INTO a USING dual ON (col3 = ? AND col1 = ? AND col2 = ?) WHEN MATCHED THEN UPDATE SET col4 = col4 + ? WHEN NOT MATCHED THEN INSERT (col1, col2, col3, col4) VALUES (?, ?, ?, ?)");
     }
 
-    @Test
```

Comment:
With these changes, this statement no longer failed to parse, so I removed the test.

Incidentally, while it's syntactically valid, it does fail this requirement (from Snowflake docs):

> Any matching or not-matching clause that omits the AND subclause (default behavior) must be the last of its clause type in the statement (e.g. a WHEN MATCHED ... clause cannot be followed by a WHEN MATCHED AND ... clause). Doing so results in an unreachable case, which returns an error.

This feels out of the scope of the validation we do here, so I haven't added anything for it, but happy to add if I'm wrong there.